sdk/python/foundation-models/system/import/import_model_into_registry.ipynb

```diff
@@ -293,50 +291,174 @@
+    "        models = list(ml_client.models.list(name=model_id))\n",
+    "        if models:\n",
+    "            max_version = (max(models, key=lambda x: int(x.version))).version\n",
+    "            does_model_exist = True\n",
+    "    except:\n",
+    "        pass\n",
+    "    finally:\n",
+    "        if does_model_exist:\n",
+    "            return latest_model\n",
+    "        else:\n",
+    "            return None\n",
+    "\n",
+    "\n",
+    "azureml_registries = [\"azureml\", \"azureml-meta\"]\n",
+    "\n",
+    "# Replace \"/\" in model id with \"-\"\n",
+    "REG_MODEL_ID = MODEL_ID.replace(\"/\", \"-\")\n",
+    "\n",
+    "does_model_exist_in_any_registry = False\n",
+    "found_registry = None\n",
+    "for registry in azureml_registries:\n",
+    "    model = does_model_exist_in_registry(credential, registry, REG_MODEL_ID)\n",
+    "    if model:\n",
+    "        does_model_exist_in_any_registry = True\n",
+    "        found_registry = registry\n",
+    "        break\n",
+    "\n",
+    "if does_model_exist_in_registry:\n",
+    "    raise Exception(\n",
+    "        f\"\"\"MODEL ALREADY EXISTS IN \"{found_registry}\" registry with name: {model.name}, version: {model.version}. You DO NOT need to import the model again.\n",
+    "                    \n",
+    "                    Please refer to the model card here: https://ml.azure.com/models/{model.name}/version/{model.version}/catalog/registry/{found_registry}. \n",
+    "                    \n",
+    "                    In case you are using SDK/CLI to submit experiments, please use the model id: {model.id} to reference the model.\"\"\"\n",
+    "    )\n",
+    "else:\n",
+    "    print(\"Model does not exist in any registry. Please proceed with importing\")"
    ]
   },
   {
-   "attachments": {},
    "cell_type": "markdown",
-   "id": "d7178a91",
+   "id": "4ca6f5a5",
    "metadata": {},
    "source": [
-    "#### Check if the model already exists in AzureML registry"
+    "#### Check that the model exists on Huggingface"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "944ea15c",
+   "id": "0baa9662",
    "metadata": {},
    "outputs": [],
    "source": [
-    "try:\n",
-    "    REG_MODEL_ID = MODEL_ID.replace(\n",
-    "        \"/\", \"-\"\n",
-    "    )  # model name in registry doesn't contain '/'\n",
-    "    models = ml_client_registry.models.list(name=REG_MODEL_ID)\n",
-    "    if models:\n",
-    "        max_version = (max(models, key=lambda x: int(x.version))).version\n",
-    "        model_version = str(int(max_version))\n",
-    "        print(\n",
-    "            f\"Model already exists in azureml with name {REG_MODEL_ID} and version {model_version}\"\n",
+    "import time\n",
+    "from typing import List\n",
+    "from huggingface_hub.hf_api import HfApi, ModelInfo, ModelFilter\n",
+    "\n",
+    "\n",
+    "def retry(times):\n",
```

Comment:
nit: is retry block needed in the notebook?

Reply:
We saw errors with HF API from the logs isn't it? Therefore kept it.

Reply:
Actually, since this is a notebook, customer can re-run it. 

Deleted retry.